ntoskrnl/ke/amd64/cpu.c

```diff
@@ -234,6 +234,60 @@ KiGetFeatureBits(VOID)
     return FeatureBits;
 }
 
+#if DBG
+VOID
+KiReportCpuFeatures(IN PKPRCB Prcb)
+{
+    ULONG CpuFeatures = 0;
+    CPU_INFO CpuInfo;
+
+    if (Prcb->CpuVendor)
+    {
+        KiCpuId(&CpuInfo, 1);
+        CpuFeatures = CpuInfo.Edx;
+    }
```

Comment:
There is `KiGetCpuVendor` in AMD64 which already returns you the CPU vendor so you do not have to pass a PCB here. So why not use that?

Reply:
On AMD64 `Prcb->CpuVendor` is being populated by `KiGetCpuVendor()` which is already has been called in phase 0 even before the Kernel Debugger initialization. Why we should call it when the value is already evaluated and cached in Prcb?